frontend/src/container/GridGraphLayout/Graph/GraphWithoutDashboard.tsx

```diff
@@ -0,0 +1,351 @@
+					},
+				],
+			}),
+		[queryResponse],
+	);
+
+	const prevChartDataSetRef = usePreviousValue<ChartData>(chartData);
+
+	const onToggleModal = useCallback(
+		(func: Dispatch<SetStateAction<boolean>>) => {
+			func((value) => !value);
+		},
+		[],
+	);
+
+	const onDeleteHandler = useCallback(() => {
+		const widgetId = isEmptyWidget ? layout[0].i : widget?.id;
+
+		featureResponse
+			.refetch()
+			.then(() => {
+				deleteWidget({ widgetId, setLayout });
+				onToggleModal(setDeleteModal);
+			})
+			.catch(() => {
+				notifications.error({
+					message: t('common:something_went_wrong'),
+				});
+			});
+	}, [
+		isEmptyWidget,
+		widget?.id,
+		layout,
+		featureResponse,
+		deleteWidget,
+		setLayout,
+		onToggleModal,
+		notifications,
+		t,
+	]);
+
+	const getModals = (): JSX.Element => (
+		<>
+			<Modal
+				destroyOnClose
+				onCancel={(): void => onToggleModal(setDeleteModal)}
+				open={deleteModal}
+				title="Delete"
+				height="10vh"
+				onOk={onDeleteHandler}
+				centered
+			>
+				<Typography>Are you sure you want to delete this widget</Typography>
+			</Modal>
+
+			<Modal
+				title="View"
+				footer={[]}
+				centered
+				open={modal}
+				onCancel={(): void => onToggleModal(setModal)}
+				width="85%"
+				destroyOnClose
+			>
+				<FullViewContainer>
+					<FullView name={`${name}expanded`} widget={widget} yAxisUnit={yAxisUnit} />
+				</FullViewContainer>
+			</Modal>
+		</>
+	);
+
+	const handleOnView = (): void => {
+		onToggleModal(setModal);
+	};
+
+	const isEmptyLayout = widget?.id === 'empty' || isEmpty(widget);
+
+	if (queryResponse.isError && !isEmptyLayout) {
+		return (
+			<span ref={graphRef}>
+				{getModals()}
+				{!isEmpty(widget) && prevChartDataSetRef && (
+					<>
+						<div className="drag-handle">
+							<WidgetHeader
+								parentHover={hovered}
+								title={widget?.title}
+								widget={widget}
+								onView={handleOnView}
+								onDelete={(): void => {}}
+								onClone={(): void => {}}
+								queryResponse={queryResponse}
+								errorMessage={errorMessage}
+								allowClone={false}
+								allowDelete={false}
+								allowEdit={false}
+							/>
+						</div>
+						<GridGraphComponent
+							GRAPH_TYPES={widget?.panelTypes}
+							data={prevChartDataSetRef}
+							isStacked={widget?.isStacked}
+							opacity={widget?.opacity}
+							title={' '}
+							name={name}
+							yAxisUnit={yAxisUnit}
+							onClickHandler={onClickHandler}
+						/>
+					</>
+				)}
+			</span>
+		);
+	}
+
+	if (prevChartDataSetRef?.labels === undefined && queryResponse.isLoading) {
+		return (
+			<span ref={graphRef}>
+				{!isEmpty(widget) && prevChartDataSetRef?.labels ? (
+					<>
+						<div className="drag-handle">
+							<WidgetHeader
+								parentHover={hovered}
+								title={widget?.title}
+								widget={widget}
+								onView={handleOnView}
+								onDelete={(): void => {}}
+								onClone={(): void => {}}
```

Comment:
we are breaking dry principle here

Reply:
Okay, I will make it optional in WidgetHeader.